databases/week3/meal_sharing.sql

```diff
@@ -0,0 +1,183 @@
+
+--Reservation
+--Get all reservations
+select * FROM reservation;
+
+--Add a new reservation
+INSERT INTO reservation(number_of_guest, meal_id, create_date,contact_phonenumber, contact_name, contact_email)
+VALUES(2, 3, now(), '349-087-033', 'Billo', 'billo@sfsd.com');
+
+--Get a reservation with any id, fx 1
+select * from reservation where id = 5;
+
+--Update a reservation with any id, fx 1. Update any attribute fx the title or multiple attributes
+UPDATE reservation
+set number_of_guest = 5, meal_id=7
+WHERE id = 3;
+
+--Delete a reservation with any id, fx 1
+DELETE FROM reservation WHERE id =5;
+
+--Review
+--Get all reviews
+SELECT * FROM review;
+
+--Add a new review
+INSERT INTO review(title, description, meal_id, stars, create_date)
+VALUES('very yumm', 'it was really delicious and one of my fav', 4, 5, now());
+
+--Get a review with any id, fx 1
+SELECT * FROM review where id = 5;
+--Update a review with any id, fx 1. Update any attribute fx the title or multiple attributes
+UPDATE review
+set title = 'Delicious'
+where id = 5;
+
+
+--Delete a review with any id, fx 1
+DELETE FROM review where id = 8;
+
+--Additional query
+--Adding different meal review and reservation
+
+INSERT INTO meal(title, description, location, `when`, max_reservations, price, create_date)
+VALUES('Rød grød med fløde', 'A salad box with Rød grød med fløde and other veggies too', 'Glostrup', NOW(), 6, 55.99, '2024-07-12'),
+('Salad-box', 'All the grrens available in one box with chicken shreds', 'Taastrup', '2024-07-15 15:45:10', 4, 80.49, '2024-07-11'),
+('Russian-salad', 'peas, broccoli and cucumber with mayos and spice up with black papper', 'Svanemøllen', '2024-07-18 01:32:00', 1, 35.79, '2024-07-03');
+
+INSERT INTO reservation(number_of_guest, meal_id, create_date, contact_phonenumber, contact_name, contact_email)
+VALUES(5, 4, now(), '577-383-689', 'avinash', 'avinash@kfjo.gk'),
+(3, 1, now(), '481-976-563', 'sushmita', 'susmita@sdfs.com'),
+(2, 3, now(), '088-490-209', 'Aarohi', 'arohi@fk.rk');
+
+insert INTO review(title, description, meal_id, stars,create_date)
+VALUES('Super', 'super delicious food', 4, 4, now()),
+('yumm-yum', 'very yummy food', 2, 3, now()),
+('awesome', 'awesome food and very delicious', 6, 5, now());
+
+--Get meals that has a price smaller than a specific price fx 90
+SELECT * from meal where price < 40;
+
+--Get meals that still has available reservations
+SELECT * FROM meal
+join reservation
+on meal_id = meal.id
+where max_reservations > number_of_guest;
+
+
+--Get meals that partially match a title. Rød grød med will match the meal with the title Rød grød med fløde
+select * from meal where title like '%Rød grød med%';
+
+--Get meals that has been created between two dates
+select * from meal where create_date BETWEEN '2024-07-03' and '2024-07-15';
+
+--Get only specific number of meals fx return only 5 meals
+select * from meal limit 3;
+
+select * from review;
+
+--Get the meals that have good reviews
+select * from meal
```

Comment:
I think the best way to get meals with good reviews it to do it by number of stars, rather than title.

Reply:
Thanks for suggestion, I did it and its really easy this way.